src/rgw/rgw_rest_s3.cc

```diff
@@ -6431,28 +6668,43 @@ int RGWSelectObj_ObjStore_S3::extract_by_tag(std::string tag_name, std::string&
   return 0;
 }
 
-int RGWSelectObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t ofs, off_t len)
+int RGWSelectObj_ObjStore_S3::extract_output_by_tag(std::string tag_name, std::string& result)
 {
-  if (len == 0) {
-    return 0;
+  result = "";
+  std::string output_tag{"OutputSerialization"};
+  size_t _qo = m_s3select_query.find("<" + output_tag + ">", 0) + output_tag.size() + 2;
```

Comment:
its not correct to do addition operation with `std::string.find`, find may return `::npos`.
there are more code lines with such an operation that should be fixed.

the input-serialization and output-serialization are both using the same tag-names, it causes wrong assignments,
the current functionality is not fit for that.

Reply:
Thanks @galsalomon66. I have updated. It might need some refactoring. Tests are passing.